game/src/info/trip.rs

```diff
@@ -593,72 +611,62 @@ fn make_bar(
+                                map.get_r(lane_detail.parent)
+                                    .get_name(app.opts.language.as_ref())
+                            )));
+                            display_txt.add(Line(format!("  Lane ID: {}", id)));
+                            display_txt.add(Line(format!(
+                                "  Lane distance: {}",
+                                lane_detail.length().describe_rounded()
+                            )));
+                            display_txt.add(Line(format!("  Average speed: {}", avg_speed)));
+                            segments.push((lane_detail.length(), Color::RED, display_txt));
+                            norm_distance = Distance::meters(0.0);
+                        } else {
                             norm_distance += lane_detail.length();
                         }
                     }
                     PathStep::Turn(id) => {
                         let turn_details = map.get_t(*id);
                         sum_phase_dist += turn_details.geom.length();
-                        let mut found = false;
-                        for (turn, delay) in &data_inter {
-                            if turn == id {
-                                segments.push((
-                                    norm_distance,
-                                    norm_color,
-                                    build_text(&msgs, &norm_distance),
-                                ));
-
-                                let mut display_txt = Text::from(Line(&p.phase_type.describe(map)));
-                                display_txt.add(Line(format!("  Intersection: {}", turn.parent)));
-                                display_txt.add(Line(format!(
-                                    "  Delay: {}",
-                                    Duration::seconds(*delay as f64)
-                                )));
-                                segments.push((
-                                    // To make sure that the hotspot isn't too small
-                                    if 0.05 < (turn_details.geom.length() / sum_phase_dist) {
-                                        turn_details.geom.length()
-                                    } else {
-                                        sum_phase_dist += sum_phase_dist * 0.05;
-                                        sum_phase_dist -= turn_details.geom.length();
-                                        sum_phase_dist * 0.05
-                                    },
-                                    Color::RED,
-                                    display_txt,
-                                ));
-                                norm_distance = Distance::meters(0.0);
-                                found = true;
-                                break;
-                            }
-                        }
-                        if !found {
+                        let delay = intersection_delays.get(id);
```

Comment:
Nit: Why not just `if let Some(delay) = intersection_delays.get(id)`?

Reply:
That is a very good point. I will change that, and any other similar occurences

Reply:
Would you change this as well?
`    let lane_speeds = &app
        .primary
        .sim
        .get_analytics()
        .lane_speed_percentage
        .get(&id);
    if let Some(lane_speeds) = lane_speeds {`
As that is a lot of consecutive function calls?